app/api/v2/views/question_views.py

```diff
@@ -187,30 +187,52 @@ def downvote(question_id):
 @question_bpv2.route('/<int:question_id>/comment', methods=['POST'])
 def comments(question_id):
     """A method to enable posting of comments based on user question """
-    question = questions.oneQuestion(question_id)
-    if question:
-        try:
-            data = request.get_json()
-        except Exception as e:
-            return make_response(jsonify({
-                "status": 400,
-                "message": "Invalid or no data sent" + e
-            }))
-        comment = data.get("comment")
-        if not comment:
+    auth_header = request.headers.get('Authorization')
+    if auth_header:
+        auth_token = auth_header.split("Bearer ")[1]
+    else:
+        auth_token = 'Bearer '
+    if auth_token:
+        response = users.verify_auth_token(auth_token)
+        if isinstance(response, str):
+            user = users.login(username=response)
+            if not user:
+                return make_response(jsonify({
+                    "status": 400,
+                    "message": "Authentication failed"
+                })), 400
```

Comment:
@a-braham you can write a decorator for your authentication so you just call it here to avoid repetition for other endpoints that require authentication